libevmasm/Assembly.cpp

```diff
@@ -342,12 +480,57 @@ Json::Value Assembly::assemblyJSON(map<string, unsigned> const& _sourceIndices)
 		}
 	}
 
-	if (m_auxiliaryData.size() > 0)
+	if (!m_auxiliaryData.empty())
 		root[".auxdata"] = toHex(m_auxiliaryData);
 
 	return root;
 }
 
+void Assembly::forEachJSONAssemblyItem(Json::Value const& _code, Assembly& _assembly)
```

Comment:
With the change I suggested above this function would get a list of assembly items. You would not have to call `assemblyItemFromJSON()` twice for each item (which is especially bad given that `assemblyItemFromJSON` has side-effects - it modifies the assembly).